ansible_collections/arista/avd/roles/eos_designs/templates/connected_endpoints/monitor-sessions.j2

```diff
@@ -0,0 +1,67 @@
+{# Leaf connected_endpoint monitor-sessions #}
+{% if monitor_sessions_data | length > 0 %}
+monitor_sessions:
+{%     for sname, group in monitor_sessions_data | groupby("name") %}
+{%         set group_roles = group | map(attribute='role', default='notdefined') | list %}
+{%         if 'source' in group_roles and 'destination' in group_roles %}
+  - name: {{ sname }}
+    sources:
+{%             for session in group | selectattr('role', 'equalto', 'source') %}
+{%                 if session.interfaces is arista.avd.defined %}
+      - name: {{ session.interfaces | join(", ") }}
+{%                 endif %}
+{%                 if session.source_settings is arista.avd.defined %}
```

Comment:
I think this should be nested under the `if session.interfaces` since without a source interface, you are writing yaml that is not part of the list entry.

Reply:
yes, thanks!!